### PR TITLE
A few small jit changes.

### PR DESCRIPTION
**Pull request by sandreenko**
Fix SPMI replay with COMPlus_JitDump, add a few new asserts, clean a few expressions.

No diffs x64 (SPC, spmi pri1 tests).

### CODE REVIEW

File: src/coreclr/src/jit/gentree.cpp
Comment (sandreenko) [inline src/coreclr/src/jit/gentree.cpp]: We use this path only in `optIsCSEcandidate`, but later `optIsCSEcandidate` doesn't know how to work with these opcodes, so it doesn't produce any diffs.

**Comment by sandreenko** [comment]
PTAL @dotnet/jit-contrib.
The main change is fix for SPMI replay, the others are just small preparations for the future PRs.

**Review by CarolEidt — APPROVED** [review]
LGTM, though it will conflict with my #811

File: src/coreclr/src/jit/gentree.cpp
Comment (mikedn) [reply src/coreclr/src/jit/gentree.cpp]: Interesting, normally the layouts used by `BLK` do not have a class handle.

File: src/coreclr/src/jit/lsraxarch.cpp
Comment (mikedn) [inline src/coreclr/src/jit/lsraxarch.cpp]: It would be more clear to say why such an indir is not expected here - `// struct typed indirs are expected to be contained` or something like that

File: src/coreclr/src/jit/lower.cpp
Comment (mikedn) [inline src/coreclr/src/jit/lower.cpp]: Seems odd to go back to direct data member access. It was, is and will always be a bad programming practice for common data structures such as IR ones.

**Comment by sandreenko** [comment]
> LGTM, though it will conflict with my #811

Thanks, I will wait until your PR is merged and then resolve the conflicts.

File: src/coreclr/src/jit/gentree.cpp
Comment (sandreenko) [reply src/coreclr/src/jit/gentree.cpp]: I usually get such BLK nodes from OBJ nodes for structs that do not contain GC pointers.
I'll check if I see them in master.

Comment (mikedn) [reply src/coreclr/src/jit/gentree.cpp]: Ah, I've managed to confuse myself. Yes, `BLK`s can have class handles now, it was my plan to have handles only in `OBJ`s but it didn't happen.

File: src/coreclr/src/jit/lower.cpp
Comment (sandreenko) [reply src/coreclr/src/jit/lower.cpp]: You are right, I forgot that `gtGetOp1` is declared for `GenTree`, not for `GenTreeOp`). Will revert that.
Thank you.

**Comment by sandreenko** [comment]
The PR was rebased and updated.

File: src/coreclr/src/jit/lsraxarch.cpp
Comment (sandreenko) [reply src/coreclr/src/jit/lsraxarch.cpp]: Thanks, done, I will change that condition soon, so it will become clearer in the next PRs as well.

**Pull request merged**
